backend/src/main/java/com/znane_groby/backend/controller/GraveyardController.java

```diff
@@ -0,0 +1,102 @@
+package com.znane_groby.backend.controller;
+
+import com.znane_groby.backend.model.Graveyard;
+import com.znane_groby.backend.repository.CityRepository;
+import com.znane_groby.backend.repository.GraveyardRepository;
+import com.znane_groby.backend.util.ItemNotFoundException;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.hateoas.CollectionModel;
+import org.springframework.hateoas.EntityModel;
+import org.springframework.hateoas.IanaLinkRelations;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.net.URISyntaxException;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+import static org.springframework.hateoas.server.mvc.WebMvcLinkBuilder.linkTo;
+import static org.springframework.hateoas.server.mvc.WebMvcLinkBuilder.methodOn;
+
+@Slf4j
+@RestController
+public class GraveyardController {
+    private final GraveyardRepository repository;
+    private final CityRepository cityRepository;
+    private final GraveyardResourceAssembler assembler;
+
+    GraveyardController(GraveyardRepository repository, GraveyardResourceAssembler assembler,CityRepository cityRepository) {
+        this.assembler = assembler;
+        this.repository = repository;
+        this.cityRepository = cityRepository;
+    }
+
+    // Aggregate root
+    @GetMapping("/graveyards")
+    CollectionModel<EntityModel<Graveyard>> all() {
+
+        List<EntityModel<Graveyard>> graveyards = StreamSupport.stream(repository.findAll().spliterator(), false)
+                .map(assembler::toModel)
+                .collect(Collectors.toList());
+        return new CollectionModel<>(graveyards,
+                linkTo(methodOn(GraveyardController.class).all()).withSelfRel());
+    }
+
+    // Single item
+    @GetMapping("/graveyards/{id}")
+    EntityModel<Graveyard> one(@PathVariable Long id) {
+        Graveyard graveyard = repository.findById(id)
+                .orElseThrow(() -> new ItemNotFoundException("/graveyards/", id));
+        log.info("graveyard.getCity() " + graveyard.getCity());
+        return assembler.toModel(graveyard);
+    }
+    // All graveyards from one city
+    @GetMapping("/graveyards/city/{id}")
+    CollectionModel<EntityModel<Graveyard>> allFromCity(@PathVariable Long id) {
+        List<EntityModel<Graveyard>> graveyards = repository.findByCity(cityRepository.findById(id)).stream()
+                .map(assembler::toModel)
+                .collect(Collectors.toList());
+        return new CollectionModel<>(graveyards,
+                linkTo(methodOn(GraveyardController.class).allFromCity(id)).withSelfRel());
+    }
+
+    @PostMapping("/graveyards")
```

Comment:
Jak chcesz robić zapis do repozytorium to pamiętaj że spring ma taki bajer jak spring data rest, który robi komplet podstawowych metod na repozytorium. To dobre do szybkiego prototypowania.
Zobacz temat `@RestResource` i pokrewne.

Reply:
właśnie o tym poczytałem i nie mogę znaleźć konkretnej odpowiedzi. Kiedy powinno się używać @RepositoryRestResource a kiedy @RestController ? Zasadniczo @Repository... jest dużo szybsze i obcina się ten powtarzający się kod, z drugiej strony @RestController zapewnia dużo większą kontrolę nad tym co robimy z danymi. wyobrażam sobie, że do niektórych klas taki @RepositoryRestResource  ma sens, a niekiedy lepszy będzie @RestController, ale czy miesza się to w projektach?